lib/rubocop/cop/style/redundant_return.rb

```diff
@@ -165,6 +167,21 @@ def message(node)
             MSG
           end
         end
+
+        def allowed_methods
+          cop_config['AllowedMethods']&.map(&:to_sym) || []
+        end
+
+        def method_name_before_of(node)
+          parent = node.parent
+          return unless %i[begin kwbegin].include?(parent.type)
+
+          node_place_index = parent.children.find_index(node)
```

Comment:
I think you need a `return unless node_place_index >= 1` or similar here. The following code won't do the right thing if the index is 0.

Reply:
Thank you!
I added `node_place_index.zero?` gurad